dev-libs/tomsfastmath/tomsfastmath-0.13.1.ebuild

```diff
@@ -0,0 +1,45 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+DESCRIPTION="TomsFastMath is a fast public domain large integer arithmetic library"
+HOMEPAGE="https://www.libtom.net/TomsFastMath/ https://github.com/libtom/tomsfastmath"
+SRC_URI="https://github.com/libtom/tomsfastmath/releases/download/v${PV}/tfm-${PV}.tar.xz"
+LICENSE="Unlicense"
+
+SLOT="0/1"
+KEYWORDS="~amd64"
+
+BDEPEND="virtual/pkgconfig"
+
+PATCHES=(
+	"${FILESDIR}/${P}-makefile-fix.patch"
+)
+
+src_compile() {
+	emake -f makefile.shared
+}
+
+src_test() {
+	emake test_standalone stest rsatest
+	./test || die "Main tests failed"
+	./stest || die "Simple test failed"
+	./rsatest || die "RSA test failed"
+}
+
+src_install() {
+	emake \
```

Comment:
I think only `LIBPATH` and `INCPATH` are actually supported by `makefile.shared`? Since they appear to support `DESTDIR`, what you probably want instead is `emake DESTDIR="${ED}" LIBPATH="/usr/$(get_libdir)" install`. That should prefix everything with `"${ED}"` for you via `DESTDIR`, and then the libpath is the only thing that needs special treatment: the other variables aren't used at all, and `INCDIR` is correct out of the box.

Reply:
It definitely felt wrong to have to provide so many -- I get the difference between PREFIX and DESTDIR now.

Reply:
It's really confusing until you get used to not only the way it's supposed to work (which is confusing enough), but also all the ways people mess up their build systems. The basic idea is that you want to install into the various `/usr/lib64`, `usr/share`, etc. subdirectories of `${ED}`. In an autotools package, `$prefix` would automatically be supported and set to `${EPREFIX}/usr` by econf (see https://dev.gentoo.org/~ulm/pms/head/pms.html#x1-12800012.3.8). The other directories (except sysconfdir...) are then relative to that. And when you run `emake DESTDIR="${D}"` as in the default `src_install`, you wind up with the paths like `${DESTDIR}${datadir} == ${DESTDIR}${prefix}/share == ${D}${EPREFIX}/usr/share == ${ED}/usr/share` in the makefile, and that's exactly what you want.

Where it gets interesting is in a hand-written makefile like this that supports only _some_ of the GNU variables. Here, `$DESTDIR` is supported, but not `$prefix`, so you need to either pass `DESTDIR="${ED}"` instead of `DESTDIR="${D}"` **or** put an `$EPREFIX` in front of `LIBPATH` and `INCPATH` manually. But in general, you're just trying to reverse engineer whatever the hand-written makefile half-supports to get it to do what it's supposed to do: configure and build stuff relative to `$EPREFIX`,  and install it relative to `$ED`.